### PR TITLE
Use ConnectionResetException with 'The client disconnected' message

### PR DESCRIPTION
**Pull request by pakrym**
https://github.com/aspnet/IISIntegration/issues/1206

### CODE REVIEW

File: test/IIS.Tests/ClientDisconnectTests.cs [hunk 1]
```diff
@@ -17,7 +18,7 @@ namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
     public class ClientDisconnectTests : LoggedTest
     {
          
-        [ConditionalFact(Skip = "See: https://github.com/aspnet/IISIntegration/issues/1075")]
+        [ConditionalFact]
```
Comment (pakrym) [inline test/IIS.Tests/ClientDisconnectTests.cs]: I re-enabled some tests that should be unaffected by pipelines bug.

File: src/Microsoft.AspNetCore.Server.IIS/Core/IO/AsyncIOOperation.cs [hunk 2]
```diff
@@ -103,7 +104,8 @@ public AsyncContinuation Complete(int hr, int bytes)
                 _result = bytes;
                 if (hr != NativeMethods.HR_OK)
                 {
-                    _exception = new IOException("Native IO operation failed", Marshal.GetExceptionForHR(hr));
+                    // Treat all errors as the client disconnect
+                    _exception = new ConnectionResetException("The client disconnected", Marshal.GetExceptionForHR(hr));
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Server.IIS/Core/IO/AsyncIOOperation.cs]: nit: The client has disconnected.

File: test/IIS.Tests/ClientDisconnectTests.cs [hunk 3]
```diff
@@ -163,6 +163,51 @@ public async Task ReaderThrowsCancelledException()
             }
         }
 
+
+
```
Comment (jkotalik) [inline test/IIS.Tests/ClientDisconnectTests.cs]: nit new lines.

File: src/Microsoft.AspNetCore.Server.IIS/Core/IO/AsyncIOOperation.cs (hunk 2 above)
Comment (pakrym) [reply src/Microsoft.AspNetCore.Server.IIS/Core/IO/AsyncIOOperation.cs]: In my defence - I copied it from System.Web.

**Pull request merged**